services/subscriber/service.go

```diff
@@ -183,20 +183,28 @@ func (s *Service) createSubscription(se subEntry, mode string, destinations []st
 	default:
 		return nil, fmt.Errorf("unknown balance mode %q", mode)
 	}
-	writers := make([]PointsWriter, len(destinations))
-	stats := make([]writerStats, len(writers))
-	for i, dest := range destinations {
+	writers := []PointsWriter{}
+	stats := []writerStats{}
+	// add only valid destinations
+	for _, dest := range destinations {
 		u, err := url.Parse(dest)
 		if err != nil {
-			return nil, err
+			s.Logger.Println("failed to parse subscription url:", dest)
```

Comment:
I first wrote it similar to your suggestion but noticed it didn't match the rest of the file. Being new to the project I just followed the pattern. There are a 11 Logger calls in the file, all of them use Println and most are sentence fragments. I've fixed my log messages.